lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/YarnClient.cs

```diff
@@ -103,6 +105,33 @@ internal YarnClient(IUrlProvider yarnRmUrlProviderUri, IRestRequestExecutor rest
                     GenerateUrlAndExecuteRequestAsync<NewApplication>(request, cancellationToken);
         }
 
+        public async Task<Application> SubmitApplicationAsync(
+            SubmitApplication submitApplication,
+            CancellationToken cancellationToken)
+        {
+            await new RemoveSynchronizationContextAwaiter();
+
+            var request = new RestRequest
+            {
+                Resource = _baseResourceString + SubmitApplication.Resource,
+                Method = Method.POST,
+                RequestFormat = DataFormat.Json,
+                JsonSerializer = new RestJsonSerializer()
+            };
+
+            request.AddBody(submitApplication);
+            var submitResponse = await GenerateUrlAndExecuteRequestAsync(request, cancellationToken);
+
+            if (submitResponse.StatusCode != HttpStatusCode.Accepted)
```

Comment:
In RestRequestExecutor.ExecuteAsync<T>, StatusCode code checking is inside the method. In RestRequestExecutor.ExecuteAsync(), it doesn't check StatusCode, instead, you check at here. Should it be consistent?